sources/niUI/src/GDRV_GLES2.cpp

```diff
@@ -4565,6 +4573,19 @@ struct cGLES2GraphicsDriver : public cIUnknownImpl<iGraphicsDriver>
         return eFalse;
       }
 
+      // XXX: We automatically reverse the culling when there is a reflection
+      // and no double sided flag is set
+      tBool isReversedCulling = eFalse;
+      if (!niFlagIs(matFlags, eMaterialFlags_DoubleSided)) {
+        const sMatrixf &drawOpMatrix = apDrawOp->GetMatrix();
+        if (niFloatIsNegative(ni::Det3x3(
```

Comment:
Are you sure that we need to compute the full matrix determinant for this? Its not enough to just look at the sign of _11,_22,_33? The determinant computation is not cheap.

Reply:
Yeah I am pretty sure. There are valid rotation matrices with negative values in the diagonal. I have test it by selecting a character in Vlk and rotating it on several axis, eventually will get negatives in the diagonal. Then some googling around corroborated my theory :P Seems like the only way to know if a matrix has a reflection is by calculating the determinant which is quite expensive. At least I can get rid of the last row and column which eliminate half of the multiplications and additions. BUT STILL, this is a expensive solution compared to the previous one which was virtually free (performance wise).

Reply:
Right. The "ReverseCulling" flag might end up being the better solution. Computing a determinant before every single drawop just for this feature is overkill and hard to justify since its such an edge case. 

The thing is that "ReverseCulling" doesnt feel super right at the Material level - although I suppose you could make the same case for DoubleSided, I think its different. "ReverseCulling" is really only when you want to mirror an object - and it must apply on the whole object, not just a single material.

So there's yet another option, which is to add it in Vlk/niPainter instead - similar to how the Wireframe flag is implemented. And have a flag in the Character entity instead. Basically in `niPainter/src/PainterRenderContext.cpp` in SetGraphicsStates you'd do `mhCurrentRasterizerStates = eCompiledStates_RS_Filled;` -> `eCompiledStates_RS_CWFilled` for the mirrored entities.

Yet another option is to table this for now and just use DoubleSided for the models that need it, are there problems with DoubleSided other than the fact that it feels a bit icky?

Reply:
No there are no problems with the Double Sided thingy other than "performance".

Reply:
Which we haven't measured... its entirely possible that the diff between DoubleSided and the clean way is < 0.1% ^^ or even not measurable. Should we close this and mark it as "docs" so we can get back to it later if we actually notice a perf issue? Then we can reuse this as a starting point.

Reply:
Yeah that's why I put "performance" in quotes :P That's fine for me.